### PR TITLE
Batch up all metric of one colelctor to be in one send

### PR DESCRIPTION
**Pull request by tsheasha**
This will help avoid lots of kernel to use switches.
Inspired from @tiras-j

### CODE REVIEW

File: src/fullerite/collector/diamond.go [hunk 1]
```diff
@@ -112,21 +112,25 @@ func (d *Diamond) Collect() {
...
+				d.Channel() <- metric
+			}
 		}
 	}
 }
 
-func (d *Diamond) parseMetric(line []byte) (metric.Metric, bool) {
-	var metric metric.Metric
-	if err := json.Unmarshal(line, &metric); err != nil {
+func (d *Diamond) parseMetric(line []byte) ([]metric.Metric, bool) {
```
Comment (dichiarafrancesco) [inline src/fullerite/collector/diamond.go]: parseMetrics instead of parseMetric? just a suggestion, you can ignore it!

Comment (tsheasha) [inline src/fullerite/collector/diamond.go]: makes sense will change it

**Comment by dichiarafrancesco** [comment]
+1

**Comment by gnufied** [comment]
Does this make a difference for individual collector? An individual collector is only going to collect in interval of pre-configured time period (for example 10s) and hence there won't be nothing much to batch, as far Python part goes.

**Comment by tsheasha** [comment]
@gnufied as this commit message states I am batching per *collector* since one collector produces one set of metrics each collection. What that means is we send out the entire collection in one batch instead of sending one metric at a time each time we call publish. So to answer your question , Yes it makes a lot of sense.

As for the collection interval I do not see how that is relevant to this commit. Please explain

**Comment by poros** [comment]
+1

**Pull request merged**